### ISSUE TITLE
Too short memory allocation leading to corruption.

### ISSUE THREAD

**Issue by jengelh** [libical/libical#295]
```
void setVObjectUStringZValue(VObject *o, const wchar_t *s)                                                              
{                                                                                                                       
    size_t size = (size_t)((uStrLen(s)+1)*2);                                                                           
    USTRINGZ_VALUE_OF(o) = (wchar_t*) dupStr((char*)s,size);                                                            
    VALUE_TYPE(o) = VCVT_USTRINGZ;                                                                                      
}                                                                                                                       
```
Why does that multiply by 2? This needs to be `(uStrLen(s) + 1) * sizeof(wchar_t)`.
Note: Replace uStrLen(s) by wcslen(s). Being a libc function that's likely also more optimized than the hand-crafted copy.

**Comment by winterz** [comment]
good catch.  ancient code.
i'm curious how you found this. code inspection?  some tool?
did it cause problems?

**Comment by jengelh** [comment]
Well, strings appeared truncated after pulling them out of the VObject again (i.e. at writeMemVObject time), says @bosim .

**Comment by winterz** [comment]
fixed in commit a6d6d7900a8087ad25daa8ef6fe7d30f5e7d9781

**Comment by bosim** [comment]
I can confirm it fixes the problem, thanks @winterz
